block/config.ml

```diff
@@ -2,9 +2,10 @@ open Mirage
 
 let main = foreign "Unikernel.Main" (console @-> block @-> job)
 
-let img = match get_mode () with
-  | `Xen -> block_of_file "xvda1"
-  | `Unix | `MacOSX -> block_of_file "disk.img"
+let img =
+  if_impl Key.is_xen
+    (block_of_file "xvda1")
+    (block_of_file "disk.img")
```

Comment:
The issue I have with get_mode is two fold:
- it's basically a hack in the engine to resolve mirage keys before other keys. I really would like `target` to be "like any other keys", but because of `get_mode`, it is not completely the case. 
- If you use it to switch implementation, it will not be shown in the description.

Note that the code is also quite easy without the `if`, you can see it [here](https://github.com/Drup/mirage/blob/9fa4a287dbe43a94ff3006451852cd4b5299fe7d/lib/mirage.ml#L410-L414).

As for the incompatibility feedback, I would prefer to solve it with https://github.com/Drup/Functoria/issues/12 in a more principled way (which would work for all the keys).